.travis.yml

```diff
@@ -1,8 +1,14 @@
 language: go
 sudo: false
-
 go:
   - 1.3
   - 1.4
   - 1.5
   - tip
+install:
+  - go get golang.org/x/tools/cmd/vet
+script:
+  - go get -t -v ./...
+  - diff -u <(echo -n) <(gofmt -d -s .)
```

Comment:
That's correct.

From man pages on `diff`:

> Exit status is 0 if inputs are the same, 1 if different, 2 if trouble.

So `diff` is used precisely to cause Travis to fail if there's a non-zero `gofmt -d` output. Just running `gofmt -d` by itself has 0 exit status and wouldn't cause Travis to fail.